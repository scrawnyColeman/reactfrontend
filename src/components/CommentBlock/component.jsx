import React from 'react';
import styled from 'styled-components';
import { colours } from '../../constants/styles';

const Container = styled.div`
    text-align: left;
`;
const StyledWrapper = styled.div`
    margin: 12px 0 12px 24px;
`;
const StyledComment = styled.div`
    max-width: 85%;
`;
const StyledCommentButtons = styled.div`
    margin: auto 0;
`;
const StyledCommentUsername = styled.span`
    color: ${colours.primary};
`;
const StyledCommentContainer = styled.div`
    display: flex;
    justify-content: space-between;
`;

const CommentBlock = ({ childComments, allComments }) => (
    <Container>
        {childComments.map((comment) => (
            <StyledWrapper key={comment.commentid} className={`parentComment-${comment.commentid}`}>
                <StyledCommentContainer>
                    <StyledComment>
                        <StyledCommentUsername>[{comment.authorusername}]:</StyledCommentUsername> {comment.comment}
                    </StyledComment>
                    <StyledCommentButtons>btn btn btn</StyledCommentButtons>
                </StyledCommentContainer>

                <CommentBlock
                    childComments={allComments.filter((searchComment) => searchComment.parentid === comment.commentid)}
                    allComments={allComments}
                    className={`childComment-${comment.commentid}`}
                />
            </StyledWrapper>
        ))}
    </Container>
);
export default CommentBlock;
