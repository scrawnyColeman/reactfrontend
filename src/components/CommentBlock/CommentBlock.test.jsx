import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import CommentBlock from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('CommentBlock component', () => {
    const allComments = [
        {
            commentid: '1',
            comment: 'This is the best test',
            parentid: '0',
            authorusername: 'Testy McTestFace',
        },
        {
            commentid: '2',
            comment: 'Im just happy to be here',
            parentid: '0',
            authorusername: 'Tester Zero',
        },
    ];
    const childComments = [
        {
            commentid: '3',
            comment: 'This is the best test',
            parentid: '1',
            authorusername: 'Testy McTestFace',
        },
        {
            commentid: '4',
            comment: 'Im just happy to be here',
            parentid: '1',
            authorusername: 'Tester Zero',
        },
    ];

    test('component matches snapshot with no comments', () => {
        const wrapper = shallow(<CommentBlock childComments={[]} allComments={[]} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('component matches snapshot with comments', () => {
        const wrapper = shallow(<CommentBlock childComments={childComments} allComments={allComments} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('comments are rendered in order', () => {
        const wrapper = shallow(<CommentBlock childComments={childComments} allComments={allComments} />);

        expect(wrapper.children().at(0).props().className).toBe('parentComment-3');
        expect(wrapper.children().at(1).props().className).toBe('parentComment-4');
    });
    test('nested comments are rendered in order', () => {
        const wrapper = shallow(<CommentBlock childComments={childComments} allComments={allComments} />);

        expect(wrapper.find(CommentBlock).at(0).props().className).toBe('childComment-3');
        expect(wrapper.find(CommentBlock).at(1).props().className).toBe('childComment-4');
    });
});
