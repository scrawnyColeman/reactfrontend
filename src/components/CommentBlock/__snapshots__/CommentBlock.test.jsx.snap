// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBlock component component matches snapshot with comments 1`] = `
<styled.div>
  <styled.div
    className="parentComment-3"
    key="3"
  >
    [
    Testy McTestFace
    ]: 
    This is the best test
    <CommentBlock
      allComments={
        Array [
          Object {
            "authorusername": "Testy McTestFace",
            "comment": "This is the best test",
            "commentid": "1",
            "parentid": "0",
          },
          Object {
            "authorusername": "Tester Zero",
            "comment": "Im just happy to be here",
            "commentid": "2",
            "parentid": "0",
          },
        ]
      }
      childComments={Array []}
      className="childComment-3"
    />
  </styled.div>
  <styled.div
    className="parentComment-4"
    key="4"
  >
    [
    Tester Zero
    ]: 
    Im just happy to be here
    <CommentBlock
      allComments={
        Array [
          Object {
            "authorusername": "Testy McTestFace",
            "comment": "This is the best test",
            "commentid": "1",
            "parentid": "0",
          },
          Object {
            "authorusername": "Tester Zero",
            "comment": "Im just happy to be here",
            "commentid": "2",
            "parentid": "0",
          },
        ]
      }
      childComments={Array []}
      className="childComment-4"
    />
  </styled.div>
</styled.div>
`;

exports[`CommentBlock component component matches snapshot with no comments 1`] = `<styled.div />`;
